### PR TITLE
Feature/summary of actions in proposal card

### PR DESCRIPTION
**Pull request by MiltonTulli**
# Description
Goal: Display a list of proposal actions (calls) inside proposal card. 

Changes
- Add jest setupTests file with global import. 
- Split proposal card styled components into it's own file. 
- Create useFilteredProposalActions hook to return proposal actions ordered. 
- Implement useFilteredProposalActions in proposalCardSummary
- Fix old router match params 
- create useGuildConfig hook to access config 
- fix useVotingResults 
- create `compact` versions of infoline components

Closes https://github.com/DXgovernance/dxvote/issues/805

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

# Checklist:

- [ ] My code follows the existing style of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any UI changes have been tested and made responsive for mobile views

# Images
Screen Shot 2022-05-11 at 16 23 39
Screen Shot 2022-05-11 at 16 23 53
Screen Shot 2022-05-11 at 16 25 07

### CODE REVIEW

File: src/Components/ProposalCard/ProposalCardActionSummary.tsx [hunk 1]
```diff
@@ -1,59 +1,25 @@
...
-    border-radius: 50%;
-  `}
-`;
-
 const ProposalCardActionSummary: React.FC<ProposalCardActionSummaryProps> = ({
-  isLoading,
+  proposal,
 }) => {
+  const { guildId } = useTypedParams();
+  const actions = useFilteredProposalActions(guildId, proposal?.id);
```
Comment (rossneilson) [inline src/Components/ProposalCard/ProposalCardActionSummary.tsx]: We should make this the same as our refactor structure so the component takes in the data as props and instead have a wrapper for this

File: src/hooks/Guilds/useVoteSummary.ts [hunk 2]
```diff
@@ -8,15 +8,16 @@ export default function useVoteSummary(
   guildId: string,
   proposalId: string
 ): number[] {
-  const { data: { totalVotes = null } = {} } = useProposal(guildId, proposalId);
+  const { data: { totalVotes = null, totalActions } = {} } = useProposal(
```
Comment (rossneilson) [inline src/hooks/Guilds/useVoteSummary.ts]: In the new optimised contracts we dont have totalActions available anymore, we need to use totalVotes.length-1

**Review by rossneilson — CHANGES_REQUESTED** [review]
Just needs the logic stripped out of component and moved to the wrapper (We can pass the component and render as a prop to avoid prop drilling) and the change to totalActions

File: src/Components/ProposalCard/ProposalCard.styled.ts [hunk 3]
```diff
@@ -0,0 +1,134 @@
+import styled from 'styled-components';
+import { Box } from 'Components/Primitives/Layout';
+import { Heading } from 'old-components/Guilds/common/Typography';
+
+export const CardWrapper = styled(Box)`
```
Comment (mprasanjith) [inline src/Components/ProposalCard/ProposalCard.styled.ts:6]: We need to base these components on a generic Card component at some point. (We have one in src/old-components/Guilds/common/Card/Card.tsx, just not too sure if its good enough to be used here.)

File: src/Components/ProposalCard/ProposalCard.test.tsx [hunk 4]
```diff
@@ -26,6 +35,7 @@ const validProps: ProposalCardProps = {
   statusProps: {
```
Comment (mprasanjith) [inline src/Components/ProposalCard/ProposalCard.test.tsx]: Can you reuse the existing ProposalStatus fixtures here? I just merged those changes, but you'll have to pull them in to fix the merge conflicts anyway.

File: src/hooks/Guilds/guild/useFilteredProposalActions.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
+import { useMemo } from 'react';
+import useProposalCalls from 'hooks/Guilds/guild/useProposalCalls';
+import {
+  DecodedAction,
+  SupportedAction,
+} from 'old-components/Guilds/ActionsBuilder/types';
+
+const isApprovalCall = (action: DecodedAction) => {
+  // TODO: improve this with better conditions. At read point decodedAction has no other things to validate than generic call && approval event in contract interface
```
Comment (mprasanjith) [inline src/hooks/Guilds/guild/useFilteredProposalActions.tsx]: We definitely should introduce a new SupportedAction type for approvals and flag it while decoding. Can do it in a different PR if it's too much work.

File: src/hooks/Guilds/guild/useFilteredProposalActions.tsx [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+};
+
+/**
+ * Importance:
+ * 1. rep minting
+ * 2. spending calls
+ * 3. transfers.
+ * 4. generic calls
+ */
+const getActionPoints = (action: DecodedAction): number => {
```
Comment (mprasanjith) [inline src/hooks/Guilds/guild/useFilteredProposalActions.tsx]: I have a hunch we might forget to update this when we introduce new SupportedAction types later. Shall we move this importance value to src/old-components/Guilds/ActionsBuilder/SupportedActions/index.tsx. We should also update the TypeScript typings to make it required.

File: src/old-components/Guilds/ActionsBuilder/SupportedActions/ERC20Transfer/ERC20TransferInfoLine.tsx [hunk 7]
```diff
@@ -9,7 +9,10 @@ import { useMemo } from 'react';
 import { FiArrowRight, FiNavigation } from 'react-icons/fi';
 import { MAINNET_ID, shortenAddress } from 'utils';
 
-const ERC20TransferInfoLine: React.FC<ActionViewProps> = ({ decodedCall }) => {
+const ERC20TransferInfoLine: React.FC<ActionViewProps> = ({
+  decodedCall,
+  compact,
```
Comment (mprasanjith) [inline src/old-components/Guilds/ActionsBuilder/SupportedActions/ERC20Transfer/ERC20TransferInfoLine.tsx:14]: Do you think it makes sense to have a different component altogether for the compact version? A `CompactInfoLine` component?

**Review by mprasanjith — COMMENTED** [review]
Great work! I've left some comments and suggestions, feel free to tackle them here or in a different PR if this PR is unwieldy.

File: src/hooks/Guilds/guild/useFilteredProposalActions.tsx (hunk 5 above)
Comment (rossneilson) [reply src/hooks/Guilds/guild/useFilteredProposalActions.tsx]: I'll make an issue for this, it does I think make sense to flag it as early as possible. We can also as Milton suggests in the comment make it more reliable to identify the spending calls

File: src/Components/ProposalCard/ProposalCardActionSummary.tsx (hunk 1 above)
Comment (MiltonTulli) [reply src/Components/ProposalCard/ProposalCardActionSummary.tsx]: I include this in proposal wrapper  and pass it down to proposalcard. Is that ok?

File: src/hooks/Guilds/useVoteSummary.ts (hunk 2 above)
Comment (MiltonTulli) [reply src/hooks/Guilds/useVoteSummary.ts]: Sure. done

File: src/Components/ProposalCard/ProposalCardActionSummary.tsx (hunk 1 above)
Comment (rossneilson) [reply src/Components/ProposalCard/ProposalCardActionSummary.tsx]: Yes that would be good for now, we might move to preferring to pass the component itself but for now yeah, pass actions as a prop via wrapper

File: src/Components/ProposalCard/ProposalCard.styled.ts (hunk 3 above)
Comment (MiltonTulli) [reply src/Components/ProposalCard/ProposalCard.styled.ts:6]: Agree, I didn't add these components. Just create a proposalcard.styled and cut all the styled components created inside proposalCard file to split concerns.I think it would be great to close this pr first.

Comment (MiltonTulli) [reply src/Components/ProposalCard/ProposalCard.styled.ts:6]: I will add a todo

File: src/Components/ProposalCard/ProposalCard.test.tsx (hunk 4 above)
Comment (MiltonTulli) [reply src/Components/ProposalCard/ProposalCard.test.tsx]: Sure. will update proposalCard tests with fixtures.

File: src/hooks/Guilds/guild/useFilteredProposalActions.tsx (hunk 6 above)
Comment (MiltonTulli) [reply src/hooks/Guilds/guild/useFilteredProposalActions.tsx]: I moved this fn to the supported actions file to hold this pointing logic. I don't know if I can add types for this unless the approval call is a supported action.  Make sense?

File: src/old-components/Guilds/ActionsBuilder/SupportedActions/ERC20Transfer/ERC20TransferInfoLine.tsx (hunk 7 above)
Comment (MiltonTulli) [reply src/old-components/Guilds/ActionsBuilder/SupportedActions/ERC20Transfer/ERC20TransferInfoLine.tsx:14]: For now, there are just small changes, like hiding some text or changing sizes, and seems ok to me for now, but let me know what you think.

**Review by rossneilson — APPROVED** [review]
Amazing work Milton!
